src/crosisdownload/mod.rs

```diff
@@ -58,37 +58,82 @@ static NO_GO: [&str; 28] = [
 
 const MAX_FILE_PARALLELISM: usize = 20;
 
+#[derive(Clone)]
 pub struct DownloadLocations {
     pub main: String,
     pub git: String,
     pub staging_git: String,
     pub ot: String,
 }
 
-enum FilePath {
-    Cont(String),
-    Break,
+pub enum DownloadStatus {
+    Full,
+    NoHistory,
+}
+
+#[derive(Clone, Copy)]
+pub struct ReplInfo<'a> {
+    pub id: &'a str,
+    pub username: &'a str,
```

Comment:
waaait a minute - a Repl doesn't have a username ☹️

Reply:
The field could be renamed to owner. I used a struct here since I didn't want a like 15 argument function (same reason why download locations are in a struct (minus the zip location))